### ISSUE TITLE
Moving RA / DEC coordinates

### ISSUE THREAD

**Issue by Hry59** [byggemandboesen/H-line-software#1]
In 12 .. 20 hours observations with four degrees intervals ( -n 50000 -i 4 .. ) I recognized that the RA coordinate adds irregularly +0.1° after ~5 .. ~18 measurement intervals. 
Independent from this moving RA the DEC values go up and down (e.g. 29.2 ..29.1 .. 29.0 ..29.1 .. 29.2) also with a changing number of intervals inbetween the different DEC values. 

I have no idea where this behaviour might come from.

**Comment by byggemandboesen** [comment]
Hi, I was not aware of this issue so thank you for raising awareness about it. I believe this issue lies in the PyEphem library and actually isn't an "issue" directly. In the calculations of celestial coordinates PyEphem include factors such as barometric pressure at the location of the observer (which is also one of the reasons why observer elevation ASL is obligatory as a parameter). This is also the reason why an observer at the North Pole pointing towards zenith will not be observing at a declination of exactly 90 degrees.

TL;DR, The issue might be a result of PyEphem's highly accurate calculations, although I will try to have a look at it!

**Comment by Hry59** [comment]
Hi,
Thank you!
barometric pressure: I had thought something like this for DEC. But this would not impact the RA calculation. 
 Btw: I like the new Plot.py with the map :)

 
 
-----Ursprüngliche Mitteilung-----
Von: Victor Boesen ***@***.***>
An: byggemandboesen/H-line-software ***@***.***>
Cc: Hry59 ***@***.***>; Author ***@***.***>
Verschickt: Fr, 14. Mai 2021 13:48
Betreff: Re: [byggemandboesen/H-line-software] Moving RA / DEC coordinates (#1)

Hi, I was not aware of this issue so thank you for raising awareness about it. I believe this issue lies in the PyEphem library and actually isn't an "issue" directly. In the calculations of celestial coordinates PyEphem include factors such as barometric pressure at the location of the observer (which is also one of the reasons why observer elevation ASL is obligatory as a parameter). This is also the reason why an observer at the North Pole pointing towards zenith will not be observing at a declination of exactly 90 degrees.TL;DR, The issue might be a result of PyEphem's highly accurate calculations, although I will try to have a look at it!—

**Comment by byggemandboesen** [comment]
Thanks! I'm glad you like it 😃 With regards to the incorrect RA I'm unsure what would be the reason for this. Do you still have your observations by any chance? If so I would like to take a look at it if possible. I've not experienced this error before and others have used the interval feature with no problems, se [here](https://twitter.com/VU2YQ/status/1371133604021997569?s=20) for example.

**Comment by Hry59** [comment]
Here is a print of a series I took recently. I'm sorry that the list is so long. I didn't  have a block editor. Changing RA decimal marked with *
Command was: python3 H-line.py -n 50000 -i 4 -c 
08.05.2021  18:13           861.064     ra=117.2,dec=-34.1.png
08.05.2021  18:29           876.115     ra=121.2,dec=-34.0.png
08.05.2021  18:45           875.518     ra=125.2,dec=-34.0.png
08.05.2021  19:01           872.385     ra=129.2,dec=-34.0.png
08.05.2021  19:17           876.438     ra=133.2,dec=-34.0.png
08.05.2021  19:33           881.623     ra=137.2,dec=-34.0.png
08.05.2021  19:49           885.006     ra=141.3,dec=-34.0.png    *
08.05.2021  20:05           891.696     ra=145.3,dec=-34.0.png
08.05.2021  20:21           882.345     ra=149.3,dec=-34.0.png
08.05.2021  20:37           878.824     ra=153.3,dec=-34.0.png
08.05.2021  20:53           876.872     ra=157.3,dec=-34.0.png
08.05.2021  21:09           878.319     ra=161.3,dec=-34.0.png
08.05.2021  21:25           880.762     ra=165.3,dec=-34.0.png
08.05.2021  21:41           893.593     ra=169.3,dec=-34.0.png
08.05.2021  21:57           895.277     ra=173.3,dec=-34.0.png
08.05.2021  22:13           880.950     ra=177.3,dec=-34.0.png
08.05.2021  22:29           876.730     ra=181.3,dec=-34.0.png
08.05.2021  22:45           858.134     ra=185.3,dec=-34.0.png
08.05.2021  23:01           862.674     ra=189.3,dec=-34.0.png
08.05.2021  23:17           857.762     ra=193.3,dec=-34.0.png
08.05.2021  23:33           859.015     ra=197.3,dec=-34.0.png
08.05.2021  23:49           858.393     ra=201.3,dec=-34.0.png
09.05.2021  00:05           896.215     ra=205.3,dec=-34.0.png
09.05.2021  00:21           913.359     ra=209.3,dec=-34.0.png
09.05.2021  00:37           904.603     ra=213.4,dec=-34.0.png    *
09.05.2021  00:53           913.026     ra=217.4,dec=-34.0.png
09.05.2021  01:09           898.886     ra=221.4,dec=-34.0.png
09.05.2021  01:25           893.876     ra=225.4,dec=-34.0.png
09.05.2021  01:41           893.390     ra=229.4,dec=-34.0.png
09.05.2021  01:57           895.105     ra=233.4,dec=-34.0.png
09.05.2021  02:13           882.268     ra=237.4,dec=-34.0.png
09.05.2021  02:29           901.849     ra=241.4,dec=-34.1.png
09.05.2021  02:45           900.124     ra=245.4,dec=-34.1.png
09.05.2021  03:01           891.973     ra=249.4,dec=-34.1.png
09.05.2021  03:17           901.477     ra=253.4,dec=-34.1.png
09.05.2021  03:33           891.553     ra=257.4,dec=-34.1.png
09.05.2021  03:49           894.901     ra=261.5,dec=-34.1.png    *
09.05.2021  04:05           886.305     ra=265.5,dec=-34.1.png
09.05.2021  04:21           907.273     ra=269.5,dec=-34.1.png
09.05.2021  04:37           894.161     ra=273.5,dec=-34.1.png
09.05.2021  04:53           888.286     ra=277.5,dec=-34.1.png
09.05.2021  05:09           898.064     ra=281.5,dec=-34.1.png
09.05.2021  05:25           891.526     ra=285.5,dec=-34.1.png
09.05.2021  05:41           891.810     ra=289.5,dec=-34.2.png
09.05.2021  05:57           896.215     ra=293.6,dec=-34.2.png    *
09.05.2021  06:13           896.861     ra=297.6,dec=-34.2.png
09.05.2021  06:29           891.108     ra=301.6,dec=-34.2.png
09.05.2021  06:45           904.349     ra=305.6,dec=-34.2.png
09.05.2021  07:01           908.290     ra=309.6,dec=-34.2.png
09.05.2021  07:17           902.374     ra=313.6,dec=-34.2.png
09.05.2021  07:33           899.291     ra=317.6,dec=-34.2.png
09.05.2021  07:49           889.025     ra=321.7,dec=-34.2.png    *
09.05.2021  08:05           901.873     ra=325.7,dec=-34.2.png
09.05.2021  08:21           902.232     ra=329.7,dec=-34.2.png
09.05.2021  08:37           888.505     ra=333.7,dec=-34.2.png
09.05.2021  08:53           891.607     ra=337.7,dec=-34.2.png
09.05.2021  09:09           883.892     ra=341.7,dec=-34.2.png
09.05.2021  09:25           892.500     ra=345.8,dec=-34.2.png    *
09.05.2021  09:41           898.389     ra=349.8,dec=-34.2.png
09.05.2021  09:57           912.471     ra=353.8,dec=-34.2.png
09.05.2021  10:13           906.576     ra=357.8,dec=-34.2.png
       62 Datei(en),     54.255.083 Bytes
BTW: While observation running the 'wait for next...' times do also change.
  
 
-----Ursprüngliche Mitteilung-----
Von: Victor Boesen ***@***.***>
An: byggemandboesen/H-line-software ***@***.***>
Cc: Hry59 ***@***.***>; Author ***@***.***>
Verschickt: Sa, 15. Mai 2021 14:08
Betreff: Re: [byggemandboesen/H-line-software] Moving RA / DEC coordinates (#1)

Thanks! I'm glad you like it 😃 With regards to the incorrect RA I'm unsure what would be the reason for this. Do you still have your observations by any chance? If so I would like to take a look at it if possible. I've not experienced this error before and others have used the interval feature with no problems, se here for example.—

**Comment by byggemandboesen** [comment]
Thanks a lot, that's very helpful! I'll have a look at it at some point to try and fix it 😉

**Comment by Hry59** [comment]
Yesterday I hopefully started a test with the new version. Today, after 16 hours I found the issue is still there:
(Changes position marked with **)
 18.05.21  15:49:00 798046 	    ra=90.8,dec=48.0.png 
 18.05.21  16:05:00 799527 	  ra=94.8,dec=48.0.png 
 18.05.21  16:21:00 801598 	  ra=98.8,dec=48.0.png 
 18.05.21  16:37:00 805465 	ra=102.8,dec=48.0.png 
 18.05.21  16:53:00 807480 	ra=106.8,dec=48.0.png 
 18.05.21  17:09:00 801372 	ra=110.8,dec=48.0.png 
 18.05.21  17:25:00 803262 	ra=114.8,dec=48.0.png 
 18.05.21  17:41:00 799578 	ra=118.8,dec=48.1.png 
 18.05.21  17:57:00 804559 	ra=122.9,dec=48.1.png **
 18.05.21  18:13:00 807459 	ra=126.9,dec=48.1.png 
 18.05.21  18:29:00 808064 	ra=130.9,dec=48.1.png 
 18.05.21  18:45:00 810290 	ra=134.9,dec=48.1.png 
 18.05.21  19:01:00 806551 	ra=138.9,dec=48.1.png 
 18.05.21  19:17:00 809136 	ra=143.0,dec=48.1.png **
 18.05.21  19:33:00 808622 	ra=147.0,dec=48.1.png 
 18.05.21  19:49:00 806295 	ra=151.0,dec=48.1.png 
 18.05.21  20:05:00 807896 	ra=155.0,dec=48.1.png 
 18.05.21  20:21:00 809402 	ra=159.0,dec=48.1.png 
 18.05.21  20:37:00 810583 	ra=163.0,dec=48.1.png 
 18.05.21  20:53:00 805021 	ra=167.1,dec=48.1.png **
 18.05.21  21:09:00 797692 	ra=171.1,dec=48.1.png 
 18.05.21  21:25:00 807645 	ra=175.1,dec=48.1.png 
 18.05.21  21:41:00 805415 	ra=179.1,dec=48.1.png 
 18.05.21  21:57:00 804573 	ra=183.1,dec=48.1.png 
 18.05.21  22:13:00 806138 	ra=187.2,dec=48.1.png **
 18.05.21  22:29:00 804337 	ra=191.2,dec=48.1.png 
 18.05.21  22:45:00 808220 	ra=195.2,dec=48.1.png 
 18.05.21  23:01:00 808257 	ra=199.2,dec=48.1.png 
 18.05.21  23:17:00 802124 	ra=203.2,dec=48.1.png 
 18.05.21  23:33:00 807289 	ra=207.3,dec=48.1.png **
 18.05.21  23:49:00 807070 	ra=211.3,dec=48.1.png 
 19.05.21  00:05:00 807635 	ra=215.3,dec=48.1.png 
 19.05.21  00:21:00 802988 	ra=219.3,dec=48.1.png 
 19.05.21  00:37:00 803141 	ra=223.3,dec=48.1.png 
 19.05.21  00:53:00 803014 	ra=227.4,dec=48.1.png **
 19.05.21  01:09:00 808531 	ra=231.4,dec=48.1.png 
 19.05.21  01:25:00 807289 	ra=235.4,dec=48.1.png 
 19.05.21  01:41:00 811656 	ra=239.4,dec=48.1.png 
 19.05.21  01:57:00 806816 	ra=243.4,dec=48.1.png 
 19.05.21  02:13:00 807200 	ra=247.4,dec=48.0.png 
 19.05.21  02:29:00 804665 	ra=251.4,dec=48.0.png 
 19.05.21  02:45:00 803708 	ra=255.5,dec=48.0.png **
 19.05.21  03:01:00 805728 	ra=259.5,dec=48.0.png 
 19.05.21  03:17:00 807466 	ra=263.5,dec=48.0.png 
 19.05.21  03:33:00 804390 	ra=267.5,dec=48.0.png 
 19.05.21  03:49:00 806928 	ra=271.5,dec=48.0.png 
 19.05.21  04:05:00 796918 	ra=275.5,dec=48.0.png 
 19.05.21  04:21:00 811137 	ra=279.5,dec=48.0.png 
 19.05.21  04:37:00 807018 	ra=283.5,dec=48.0.png 
 19.05.21  04:53:00 809660 	ra=287.6,dec=48.0.png **
 19.05.21  05:09:00 808160 	ra=291.6,dec=48.0.png 
 19.05.21  05:25:00 807517 	ra=295.6,dec=48.0.png 
 19.05.21  05:41:00 807752 	ra=299.6,dec=47.9.png 
 19.05.21  05:57:00 811335 	ra=303.6,dec=47.9.png 
 19.05.21  06:13:00 806277 	ra=307.6,dec=47.9.png 
 19.05.21  06:29:00 804892 	ra=311.6,dec=47.9.png 
 19.05.21  06:45:00 808254 	ra=315.6,dec=47.9.png 
 19.05.21  07:01:00 810015 	ra=319.6,dec=47.9.png 
 19.05.21  07:17:00 809004 	ra=323.6,dec=47.9.png 
 19.05.21  07:33:00 802927 	ra=327.6,dec=47.9.png 
 19.05.21  07:49:00 808800 	ra=331.6,dec=47.9.png 
 19.05.21  08:05:00 808065 	ra=335.6,dec=47.9.png 
 19.05.21  08:21:00 810814 	ra=339.6,dec=47.9.png 
 19.05.21  08:37:00 809210 	ra=343.6,dec=47.9.png 
 19.05.21  08:53:00 809620 	ra=347.6,dec=47.9.png 
 19.05.21  09:09:00 805595 	ra=351.6,dec=47.9.png 
 19.05.21  09:25:00 809325 	ra=355.6,dec=47.9.png 
 19.05.21  09:41:00 811897 	ra=359.6,dec=47.9.png 
 19.05.21  09:57:00 807130 	    ra=3.7,dec=47.9.png **
 19.05.21  10:13:00 803829 	    ra=7.7,dec=47.9.png 
 19.05.21  10:29:00 803817 	  ra=11.7,dec=47.9.png 
 19.05.21  10:45:00 807195 	  ra=15.7,dec=47.9.png 
 19.05.21  11:01:00 801887 	  ra=19.7,dec=47.9.png 
 19.05.21  11:17:00 807545 	  ra=23.7,dec=47.9.png 
 19.05.21  11:33:00 804891 	  ra=27.7,dec=47.9.png 
 19.05.21  11:49:00 797754 	  ra=31.7,dec=47.9.png 
               77 Datei(en),     61.268.361 Bytes   
(The zero set seconds of the time stamps are probably caused by Linux)
I'm sorry that there's no better news.

**Comment by byggemandboesen** [comment]
Hi, I'm still looking into this problem but quite busy due to exams and etc. The recent commit wasn't meant to fix the problem, but I'll let you know inside this issue when the problem will be solved 😉 Thanks for the very useful data! It really helps a lot.

**Comment by Hry59** [comment]
Take your time. The good thing is that the universe can't run away :)
 
 
 
-----Ursprüngliche Mitteilung-----
Von: Victor Boesen ***@***.***>
An: byggemandboesen/H-line-software ***@***.***>
Cc: Hry59 ***@***.***>; Author ***@***.***>
Verschickt: Mi, 19. Mai 2021 21:23
Betreff: Re: [byggemandboesen/H-line-software] Moving RA / DEC coordinates (#1)

Hi, I'm still looking into this problem but quite busy due to exams and etc. The recent commit wasn't meant to fix the problem, but I'll let you know inside this issue when the problem will be solved 😉 Thanks for the very useful data! It really helps a lot.—

**Comment by byggemandboesen** [comment]
Hi, the issue proved itself to be harder to fix than firstly anticipated due to code execution time/delay, so I could never get the code to execute at exact times. In the end I managed to fix the bug another way, so it should be gone by now! Thanks for your extremely thorough explanation of the issue and please let me know in the next couple of days if you have any problems with the fix. Otherwise I will close this issue and consider it fixed 😃

**Comment by Hry59** [comment]
Hi, you've got it solved. Running a 24 hour test produced constant values in RA and Dec.Lots of thanks for your work!
 
 
 
-----Ursprüngliche Mitteilung-----
Von: Victor Boesen ***@***.***>
An: byggemandboesen/H-line-software ***@***.***>
Cc: Hry59 ***@***.***>; Author ***@***.***>
Verschickt: Sa, 17. Jul. 2021 17:52
Betreff: Re: [byggemandboesen/H-line-software] Moving RA / DEC coordinates (#1)

Hi, the issue proved itself to be harder to fix than firstly anticipated due to code execution time/delay, so I could never get the code to execute at exact times. In the end I managed to fix the bug another way, so it should be gone by now! Thanks for your extremely thorough explanation of the issue and please let me know in the next couple of days if you have any problems with the fix. Otherwise I will close this issue and consider it fixed 😃 —

**Comment by byggemandboesen** [comment]
I'm glad it's working. Thank you for your support!

**Comment by Hry59** [comment]
THANK YOU for doing all that great work!
 
 
 
-----Ursprüngliche Mitteilung-----
Von: Victor Boesen ***@***.***>
An: byggemandboesen/H-line-software ***@***.***>
Cc: Hry59 ***@***.***>; Author ***@***.***>
Verschickt: Di, 20. Jul. 2021 20:45
Betreff: Re: [byggemandboesen/H-line-software] Moving RA / DEC coordinates (#1)

I'm glad it's working. Thank you for your support!—
